zomes/chat/src/entries/message/handlers.rs

```diff
@@ -163,39 +170,115 @@ fn get_messages(links: Vec<Link>) -> ChatResult<Vec<MessageData>> {
     Ok(messages)
 }
 
-/// Add the time from the Date type to this path
-fn add_time_path(path: Path, date: Date) -> ChatResult<Path> {
-    let Date { year, month, day } = date;
+/// Add the chunk index from the Date type to this path
+fn add_chunk_path(path: Path, chunk: u32) -> ChatResult<Path> {
     let mut components: Vec<_> = path.into();
 
-    // Add each part of the date as a component
-    // so our path will be `category:channel_id:year:month:day`.
-    // For example `General:3289hdf9823h92:2020:09:17` might be a
-    // path for all the messages on the 17th of September.
-
-    components.push(year.into());
-    components.push(month.into());
-    components.push(day.into());
+    components.push(format!("{}", chunk).into());
     Ok(components.into())
 }
 
-/// Add the current date to the path.
-/// This works the same as the above but uses current
-/// system time. Note that system time is unreliable but
-/// so this would require more thought in a production app.
-fn add_current_time_path(path: Path) -> ChatResult<Path> {
-    use chrono::Datelike;
+pub(crate) fn signal_users_on_channel(signal_message_data: SignalMessageData) -> ChatResult<()> {
+    let me = agent_info()?.agent_latest_pubkey;
+
+    let path: Path = signal_message_data.channel_data.channel.chatters_path();
+    let hour_path = add_current_hour_path(path.clone())?;
+    hour_path.ensure()?;
+    signal_hour(hour_path, signal_message_data.clone(), me.clone())?;
+    let hour_path = add_current_hour_minus_n_path(path, 1)?;
+    hour_path.ensure()?;
+    signal_hour(hour_path, signal_message_data, me)?;
+    Ok(())
+}
+
+fn signal_hour(
+    hour_path: Path,
+    signal_message_data: SignalMessageData,
+    me: AgentPubKey,
+) -> ChatResult<()> {
+    let chatters = get_links(hour_path.hash()?, None)?.into_inner();
+    debug!(format!("num online chatters {}", chatters.len()));
+    for tag in chatters.into_iter().map(|l| l.tag) {
+        let agent = tag_to_agent(tag)?;
+        if agent == me {
+            continue;
+        }
+        debug!(format!("Signaling {:?}", agent));
+        call_remote(
```

Comment:
It looks to me like without the failure case for `signal_hour` is going to be very expensive.  Let's say there are 100 agents in a chat channel, and all of a sudden your connection drops, you type a message and your zome is going to try and send 100 consecutive call_remotes that will all have to time out before the zome `signal_users_on_channel` zome call returns.  The user experience on that zome call is a hung conductor.  What we need is something like a `call_remote_multi` that you can pass in a `Vec` of agents, and all the separate calls can run async.